cheroot/test/test_server.py

```diff
@@ -439,3 +441,52 @@ def many_open_sockets(request, resource_limit):
         # Close our open resources
         for test_socket in test_sockets:
             test_socket.close()
+
+
+@pytest.mark.parametrize(
+    ('minthreads', 'maxthreads', 'error'),
+    (
+        (-1, -1, ValueError),
+        (-1, 0, ValueError),
+        (-1, 1, ValueError),
+        (-1, 2, ValueError),
+        (0, -1, ValueError),
+        (0, 0, ValueError),
+        (0, 1, ValueError),
+        (0, 2, ValueError),
+        (1, -1, None),
+        (1, 0, ValueError),
+        (1, 1, None),
+        (1, 2, None),
+        (2, -1, None),
+        (2, 0, ValueError),
+        (2, 1, ValueError),
+        (2, 2, None),
+    ),
+)
+def test_threadpool_threadnum_validation(minthreads, maxthreads, error):
+    kwargs = dict(
+        server=None,
+        min=minthreads,
+        max=maxthreads,
+    )
+    if error is None:
```

Comment:
The tests shouldn't be branchy, logic in tests should be avoided as much as possible.
Besides, I see two distinct tests here — negative and positive. This is a clear hint that they should reside in separate test functions with separate names clearly stating what they are testing — this is also useful in the test report output.

Reply:
I pulled the test apart into 2 tests - one for the errors (`test_threadpool_invalid_threadrange`), and one that ensures that valid min/max values are properly set (`test_threadpool_threadrange_set`). This removed the branching, the extra variable, and the need for an `ids` function.